hooks/pre-command

```diff
@@ -56,6 +56,10 @@ if [[ -n "${BUILDKITE_PLUGIN_CACHE_KEY:-}" ]]; then
 
   if [ -f "$BK_CACHE_BASEDIR/lib/backends/$BK_CACHE_BACKEND.bash" ]; then
     . "$BK_CACHE_BASEDIR/lib/backends/$BK_CACHE_BACKEND.bash"
+    if [[ "${BUILDKITE_PLUGIN_CACHE_CONTINUE_ON_RESTORE_ERROR:-false}" =~ (true|on|1) ]]; then
+      info "Continuing on any errors occured during cache restore as per instructed by continue_on_restore_error"
+      set +e
+    fi
     restore
```

Comment:
I think we want to `set -e` again after the `restore` to make sure this flag isn't acting on more than intended. It doesn't look like it would matter right now, but code might be added to the end of this script in the future and this would be easy to miss at that point.